File: hw2/Codes/Oztunc_Zeynep.lua
--example of an unconditional exit
local loopVar=10
while( loopVar>=10 )
do
   loopVar=loopVar-1
   break
end

print("The value of loopVar after loop is: ".. loopVar)

--using break in enclosing loops    
--outer loop
for i=0,5 do
  print("Inside the outer loop, i:"..i); 
  print("------------------------------");

    --inner loop
    for j=6,8 do                  
         --this break statement quits inner loop
         if i == 2 then
            print("Exiting from the inner for loop")
            break
          end

        print("Inside the inner loop, j ".. j)
    end


    print("\n")
end

--example of goto
  print("Using goto to exit")
  for i=0,5 do
  print("Inside the outer loop, i:"..i); 
  print("------------------------------");

      for j=6,8 do                  
      print("Inside the inner loop, j: ".. j)

          --this goto statement jumps to Exit label
          if i == 2 then
             print("Exiting from the inner for loop")
             goto Exit
           end
    end
    print("\n")
  end
  :: Exit ::

   

File: hw2/Codes/Oztunc_Zeynep.dart
void main() {
  
   print("Example  of break as an integral part");
   var loopVar = 10; 
   
   while(loopVar >=1) { 
      loopVar--; 
      break;
   } 
   print("The value of loopVar after the loop is: ${loopVar}"); 
  
  print("\n Example  of continue");
   loopVar = 10; 
   
   while(loopVar >=1) { 
      loopVar--; 
      continue;
   } 
   print("The value of loopVar after the loop is: ${loopVar}"); 

 print("\n");
 print("Example  of break in an enclosing loop");
  
 //outer label
 outer: 
    for (var i = 0; i < 5; i++) { 
      print("Inside the outer loop, i: ${i}"); 
      print("------------------------------");
     
      //inner label 
      inner: 
        for (var j = 6; j < 9; j++) { 
         
          //this break statement quits the outer loop
          if (i == 2) {
            print("Exiting from outer");
            break outer; 
          }
          print("Inside the inner loop, j:  ${j}"); 
        }
        print("\n");

  } 
  
  print("\n");
  print("Example  of continue in an enclosing loop"); 
  outer: 
   for (var i = 0; i < 5; i++) { 
      print("Inside the outer loop, i: ${i}"); 
      print("------------------------------");
     
      //inner label 
      inner: 
        for (var j = 6; j < 9; j++) { 
         

          //this break statement quits the outer loop
          if (i == 2) {
            print("Continuing from outer \n");
            continue outer; 
          }
          print("Inside the inner loop, j:  ${j}"); 

        }
        print("\n");
   } 

}

